src/dropdowns/dropdown.ts

```diff
@@ -0,0 +1,136 @@
+import {Directive, Input, Output, EventEmitter, HostBinding, HostListener, ElementRef, OnInit, OnDestroy, ContentChildren, QueryList, Renderer} from 'angular2/core';
+import {NglDropdownItem} from './dropdown-item';
+import {toBoolean} from '../util/util';
+
+const openEventEmitter = new EventEmitter<any>(false);
+
+@Directive({
+  selector: '[nglDropdown]',
+})
+export class NglDropdown implements OnInit, OnDestroy {
+  @Input('open') set isOpen(isOpen: boolean | string) {
+    isOpen = toBoolean(isOpen);
+     if (isOpen) {
+       this._subscribeToPageClickEvents();
+       this.handlePageClickEvents = false;
+       setTimeout(() => this.handlePageClickEvents = true);
+     }
+    this._isOpen = <boolean>isOpen;
+  }
+  get isOpen() {
+    return this._isOpen;
+  }
+  @Input() handlePageEvents = true;
+  @ContentChildren(NglDropdownItem) items: QueryList<NglDropdownItem> = new QueryList<NglDropdownItem>();
+  @Output('openChange') isOpenChange = new EventEmitter<boolean>(false);
+  @HostBinding('class.slds-is-open')
+  @HostBinding('attr.aria-expanded')
+  get __isOpen() {
+    return this.isOpen;
+  }
+
+  private handlePageClickEvents = true;
+  private _isOpen = false;
+  private openEventSubscription: any;
+  private triggerFocusEventEmitter = new EventEmitter(false);
+  private clickEventUnsubscriber: Function = null;
+
+  @HostListener('keydown.esc', ['$event'])
+  @HostListener('keydown.tab')
+  onKeydownClose($event: Event) {
+    this.toggle(false);
+    if ($event) {
+      this.triggerFocusEventEmitter.emit(null);
+    }
+  }
+  @HostListener('keydown.arrowdown', ['$event'])
+  onKeydownFocusNext($event: Event) {
+    $event.preventDefault();
+    this.focusItem('next');
+  }
+  @HostListener('keydown.arrowup', ['$event'])
+  onKeydownFocusPrevious($event: Event) {
+    $event.preventDefault();
+    this.focusItem('previous');
+  }
+
+  constructor(private elementRef: ElementRef, private renderer: Renderer) {}
+
+  ngOnInit() {
+    this.openEventSubscription = openEventEmitter.subscribe(this.handleDropdownOpenEvent.bind(this));
+  }
+
+  ngOnDestroy() {
+    this.openEventSubscription.unsubscribe();
+    this._unsubscribeFromPageClickEvents();
+  }
+
+  subscribeTrigger(callback: Function) {
+    this.triggerFocusEventEmitter.subscribe(callback);
+  }
+
+  toggle(toggle: boolean = !this.isOpen, focus: boolean = false) {
+    if (toggle === this.isOpen) {
+      return;
+    }
+    this.isOpenChange.emit(toggle);
+    if (toggle) {
+      this._subscribeToPageClickEvents();
```

Comment:
Is this needed here, since you subscribe on page click events when open input is changed? Also the fact that you `emit` to toggle open doesn't mean it will actually happen, in order to subscribe to this.